src/util/helperFunctions.ts

```diff
@@ -0,0 +1,44 @@
+export function regexGroupToArray(basis: string, pattern: RegExp, group: number): string[] {
+	const groupContent: string[] = [];
+	let matchGroups: string[] | null;
+
+	while ((matchGroups = pattern.exec(basis)) !== null) {
+		groupContent.push(matchGroups[group]);
+	}
+
+	return groupContent;
+}
+
+export function kahnAlgorithm(graph: Set<string[]>): string[] {
+	const order: string[] = [];
+	const vertexesWithNoInEdge: Set<string> = getVertexesWithNoInEdge(graph);
+
+ while (vertexesWithNoInEdge.size !== 0) {
+		vertexesWithNoInEdge.forEach((vertex: string) => {
+			vertexesWithNoInEdge.delete(vertex);
+			order.push(vertex);
+		 graph.forEach((edge: string[]) => {
+				if (edge[0] === vertex) { graph.delete(edge); }
+				vertexesWithNoInEdge.add(edge[1]);
+			});
+		});
+	}
+ if (graph.size !== 0 ) { throw new Error("Cyclic Import"); }
```

Comment:
You could return that residual graph here to let the user know what the circle is.